frontend/app/actions/RegistrationUri.scala

```diff
@@ -33,10 +34,11 @@ object RegistrationUri {
     controllers.routes.Joiner.joinFriend().path -> "FRI"
   )
 
-  def parse(request: RequestHeader) = {
-    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+  def parse(request: RequestHeader): String = {
     val campaignCode = extractCampaignCode(request.headers.get(REFERRER_HEADER), request.path)
-    redirectUrl.concat(s"&INTCMP=$campaignCode")
+    val returnUri: String = (request.uri ? ("INTCMP" -> campaignCode))
```

Comment:
Ok renamed the variables (actually, got rid of one from your comment below): 019179d.